Submitted By:            Igor Živković <contact@igor-zivkovic.from.hr>
Date:                    2014-03-31
Initial Package Version: 2.7.6
Upstream Status:         Unknown
Origin:                  https://github.com/yyuu/pyenv/issues/149
Description:             Fixes building readline module with Readline-6.3

diff -Naur Python-2.7.6.orig/Modules/readline.c Python-2.7.6/Modules/readline.c
--- Python-2.7.6.orig/Modules/readline.c	2013-11-10 08:36:41.000000000 +0100
+++ Python-2.7.6/Modules/readline.c	2014-03-31 08:41:09.481527357 +0200
@@ -911,17 +911,17 @@
     rl_bind_key_in_map ('\t', rl_complete, emacs_meta_keymap);
     rl_bind_key_in_map ('\033', rl_complete, emacs_meta_keymap);
     /* Set our hook functions */
-    rl_startup_hook = (Function *)on_startup_hook;
+    rl_startup_hook = on_startup_hook;
 #ifdef HAVE_RL_PRE_INPUT_HOOK
-    rl_pre_input_hook = (Function *)on_pre_input_hook;
+    rl_pre_input_hook = on_pre_input_hook;
 #endif
     /* Set our completion function */
-    rl_attempted_completion_function = (CPPFunction *)flex_complete;
+    rl_attempted_completion_function = flex_complete;
     /* Set Python word break characters */
     completer_word_break_characters =
         rl_completer_word_break_characters =
         strdup(" \t\n`~!@#$%^&*()-=+[{]}\\|;:'\",<>/?");
-        /* All nonalphanums except '.' */
+        /* All nonalphanums except '.' test */
 
     begidx = PyInt_FromLong(0L);
     endidx = PyInt_FromLong(0L);
